src/components/List/ItemListMapper.tsx

```diff
@@ -0,0 +1,44 @@
+import { List } from "@/pages/list";
+import { sortByAlphabet, sortByCategory } from "./SortBySwitches";
+
+type ItemListMapperProps = {
+  itemList: List;
+  sortBy: string;
+};
+
+export function ItemListMapper(props: ItemListMapperProps) {
+  let sectionName = "";
+  let sorted: List = [];
+
+  switch (props.sortBy) {
+    case "date":
+      sorted = props.itemList;
+      break;
+    case "category":
+      sorted = sortByCategory(props.itemList);
+      break;
+    case "alphabetical":
+      sorted = sortByAlphabet(props.itemList);
+      break;
+  }
+  return (
+    <>
+      {sorted.map((product) => {
+        let nameSection = false;
+
+        if (props.sortBy === "category") {
```

Comment:
try to avoid nesting of if statements if you can handle both conditions in one:
`(props.sortBy === "category") && (product.category !== sectionName)`